specs/HostResourceMapping.md

```diff
@@ -0,0 +1,103 @@
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
+webview2->Navigate("https://app-file.invalid/index.html");
+```
+
+## .Net and WinRT
+```c#
+webView.CoreWebView2.AddVirtualHostNameToFolderMapping(
+    "app-file.invalid", ".", CoreWebView2HostResourceAccessKind.DenyCors);
+webView.Navigate("https://app-file.invalid/index.html");
+```
+
+# Remarks
+Choose a virtual host name that will not be used by real sites.  
+If you own a domain (like example.com), an option is to use a sub domain reserved for the app (like my-app.example.com).  
+If you don't own a domain, [RFC 2606](https://tools.ietf.org/html/rfc2606#section-2) has reserved several top level domain that would
+not be used by real sites, like .example, .test, and .invalid.
+
+Give only minimal cross orign access neccessary to run the app. If there is no need to access local resources
+from other sites, use COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+
+```IDL
+/// Kind of cross origin resource access allowed for host resources during download.
+/// Note that other normal access checks like same origin DOM access check and [Content
+/// Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) still apply.
+[v1_enum]
+typedef enum COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND {
+  /// All cross origin resource access is denied, including normal sub resource access
+  /// as src of a script or image element.
+  COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY,
+
+  /// All cross origin resource access is allowed, including accesses that are
+  /// subject to Cross-Origin Resource Sharing(CORS) check. The behavior is similar to
+  /// a web site sends back http header Access-Control-Allow-Origin: *.
+  COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_ALLOW,
+
+  /// Cross origin resource access is allowed for normal sub resource access like
+  /// as src of a script or image element, while any access that subjects to CORS check
+  /// will be denied.
+  /// See [Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+  /// for more information.
+  COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS,
+} COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND;
+
+interface ICoreWebView2_2 : ICoreWebView2 {
+
+  /// Add a host name mapping for host resources in a folder.
+  /// After adding the mapping, the app can then use http or https urls with the specified hostName as host name
+  /// of the urls to access files in the local folder specified by folderPath.
```

Comment:
It would be nice to make it explicit that this applies to top level document navigations (like you've shown in the example) as well as any subresources referenced by documents from a virtual host name and otherwise.